### DISCUSSION TITLE
Testing model events

### DISCUSSION THREAD

**Discussion by domsumac** [spatie/laravel-activitylog#1166] [Q&A]
Hi All,

I'm currently writing some tests for activity logging, for model deletions I'm utilising `getActivitylogOptions()`. The logging itself functions fine when working locally, though when writing tests, model deletion does not seem to log anything though create/update tests are passing fine. Here's an example:

```
it('should create a log when an admin is deleted', function() {

  $administrator = Administrator::where('first_name', 'First Name')->first();

  $administrator->delete();

  //this passes
  $this->assertSoftDeleted('administrators', [
    'first_name' => $administrator->first_name,
    'last_name' => $administrator->last_name,
    'email' => $administrator->email,
  ]);

  //this fails
  $this->assertDatabaseHas('activity_log', [
    'id' => 1,
    'description' => 'admin deleted',
    'subject_type' => 'App\\Models\\Administrator',
    'subject_id' => $administrator->id,
    'causer_type' => 'App\\Models\\Administrator',
    'event' => 'deleted'
  ]);
```

Dumping out the DB `activity_log` will also an empty collection. This issue is happening in other deletions but they also involve firing a chain of jobs, this is just handled by a simple `$administrator->delete()` in the admin controller, which has lead me to think it's something to do with the relationship between testing and model events.

**Comment by PierreMaraisZA** [comment]
did you end up resolving this?
